routes/api/contacts.js

```diff
@@ -1,25 +1,74 @@
-const express = require('express')
+const express = require("express");
+const {
+  listContacts,
+  getContactById,
+  addContact,
+  removeContact,
+  updateContact,
+} = require("../../models/contacts");
 
-const router = express.Router()
+const validate = require("./../../validator/validator");
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const contactsRouter = express.Router();
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+contactsRouter.get("/contacts", async (__, res) => {
+  const contacts = await listContacts();
+  res.status(200).json(contacts);
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+contactsRouter.get("/contacts/:Id", async (req, res) => {
+  const { Id } = req.params;
+  const contact = await getContactById(Id);
+  if (contact) {
+    res.status(200).json(contact);
+  } else {
+    res.status(404).json({ message: `Not found` });
+  }
+});
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+contactsRouter.post("/contacts",validate.contactValid, async (req, res) => {
+  const newContact = await addContact(req.body);
+  res.json({ status: "success", code: 201, data: { newContact } });
+});
 
-module.exports = router
+contactsRouter.delete("/contacts/:Id", async (req, res) => {
+  const { Id } = req.params;
+  const contact = await removeContact(Id);
+  if (contact) {
+    res.status(200).json({ message: "contact deleted" });
+  } else {
+    res.status(404).json({ message: "Not found" });
+  }
+});
+
+contactsRouter.put("/contacts/:Id",validate.contactValid, async (req, res) => {
```

Comment:
validate.contactValid mogłaby jakoś zająć się tym że jeżeli nie ma danych w body to powinna przerwać działanie żebyś nie musiał tego jeszcze raz robić w linii 48